api/graphql.go

```diff
@@ -62,14 +61,13 @@ func BuildCorrectnessQuery(ownerName string, repoName string) (query map[string]
 	return correctnessQuery
 }
 
-func GetCorrectnessFactors(ownerName string, repoName string) (watchers int64, stargazers int64, totalCommits int64, err error) {
+func GetCorrectnessFactors(url string) (watchers int64, stargazers int64, totalCommits int64, err error) {
+	ownerName, repoName, token, err := validateInput(url)
+	if err != nil {
+		return 0, 0, 0, fmt.Errorf("GetCorrectnessFactors: Error on validate input")
```

Comment:
should this return -1's instead to indicate an error state?

Reply:
other than the faileing checks, I approve

Reply:
-1 is used to say that the metric was not implemented.